cdmtaskservice/job_state.py

```diff
@@ -198,3 +198,16 @@ async def get_job(
             msg = f"Admin user {user} accessed {job.user}'s job {job_id}"
         logging.getLogger(__name__).info(msg, extra={logfields.JOB_ID: job_id})
         return job
+    
+    async def list_jobs(self, user: str, limit: int = 1000) -> list[models.JobPreview]:
+        """
+        List jobs for a user.
+        
+        user - the user requesting jobs.
+        limit - the maximum number of jobs to return between 1 and 1000.
+        """
+        # mostly a pass through method
+        limit = 1000 if limit is None else limit
```

Comment:
The default value for `limit` is 1000 and type of `limit` is `init`, but then there's also logic handling `limit` is None?

Reply:
Yep, there's nothing preventing someone from passing None.

I'll grant you that there's also nothing preventing someone from passing a non-int either, but that seems like overkill to me. Adding type checking to every argument is non-pythonic and would add a lot of code that is almost never needed. Passing None seems much more likely to happen.

This kind of stuff is why I prefer typed languages...